scripts/anonymization/README.md

```diff
@@ -0,0 +1,220 @@
+
+The anonymization module allows to apply privacy mechanisms such as differential privacy or column faking to the data.
+The system will pull data from the JDBC connection, anonymize the data and push it back to the DBMS by creating a new table.
+
+## Config
+
+### Default Parameters
+
+The default parameters must be added inside of the `<table>`-tag. (See below)
+
+| Name        | Default | Possible Value                      | Description                                                                                  |
+| ----------- | ------- | ----------------------------------- | -------------------------------------------------------------------------------------------- |
+| name        | -       | Any string                          | The name of the table                                                                        |
+| epsilon     | 1.0     | Any value > 0                       | The privacy budget. Higher values will decrease the privacy guarantee                        |
+| pre_epsilon | 0.5     | Any value between 0 and epsilon     | The privacy budget spent on preprocessing of the data. Only necessary for continuous columns |
+| algorithm   | mst     | One of: `mst,aim,dpctgan,patectgan` | The differential privacy mechanism applied to the data                                       |
+
+```xml
+<anonymization>
+    <table
+        name="tablename"
+        epsilon="1.5"
+        pre_epsilon="0.75"
+        algorithm="patectgan"
+    />
+</anonymization>
+```
+
+### Column Information
+
+The columns of a table can be split into four categories:
+
+1. `drop` - Columns that are omitted from the DP-mechanism
+2. `categorical` - Columns that contain categorical data
+3. `continuous` - Columns of data on a continuous domain. Only numerical columns
+4. `ordinal` - Columns of ordinal meaning. Typically numerical columns with low amount of distinct values
+
+The names of the columns must be given as text. Multiple column names must be separated by comma without additional whitespace.
+
+The anonymization process will automatically try to infer the category best suitable category of each column. Sometimes this leads to undesired results or is unable to process the column because of other factors. To counter this, each column can be defined by hand with the suitable tags.
+
+---
+
+**Drop the ID-column and anonymize automatically:**
+
+```xml
+<anonymization>
+    <table name="item">
+        <drop>i_id</drop>
+    </table>
+</anonymization>
+```
+
+The ID-column will be removed from the anonymization process and be added back to the data in its original form! This is useful for primary key columns or columns that contain only NULL values.
+
+---
+
+**Define the type of each column individually:**
+
+```xml
+<anonymization>
+    <table name="item">
+        <categorical>i_name,i_data,i_im_id</categorical>
+        <continuous>i_price</continuous>
+        <ordinal>i_id</ordinal>
+    </table>
+</anonymization>
+```
```

Comment:
Alternative XML schema:

```xml
<anonymization>
  <table name="t">
    <columns>
      <column name="c1" type="continuous" />
      <column name="c2" type="continuous" />
      <column name="c2" type="ordinal" />
      <column name="c3" type="categorical" />
      <column name="c4" type="passthru" />
  </table>
```

Reply:
Basically, avoid sub parsing in another language (csv) and just use xml for the whole thing in some way.

Reply:
Also, then all your "fine tuning" configs below can get merged into this rather than having two separate sections.

Reply:
Changed Schema to avoid sub parsing. See README